tests/integration/envrctestcase.rs

```diff
@@ -0,0 +1,230 @@
+//! Implement a wrapper around setup and tear-down of Direnv-based test
+//! cases.
+
+use direnv::DirenvEnv;
+use lorri::{
+    build_loop::{BuildError, BuildLoop, BuildResults},
+    ops::direnv,
+    project::Project,
+    roots::Roots,
+};
+use std::collections::HashMap;
+use std::ffi::{OsStr, OsString};
+use std::fmt::Display;
+use std::fs::{create_dir, File};
+use std::io::BufWriter;
+use std::io::Write;
+use std::os::unix::ffi::OsStrExt;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use tempfile::{tempdir, TempDir};
+
+fn find_program<S: Display + AsRef<OsStr>>(program: S) -> PathBuf {
+    // Find programs: the environment variables in tests very likely
+    // makes PATH bogus.
+    let output = Command::new("bash")
+        .args(&["-c", "which \"$1\"", "--"])
+        .arg(&program)
+        .output()
+        .expect(&format!(
+            "Failed to execute «bash -c 'which {}'»",
+            &program
+        ));
+
+    assert!(
+        output.status.success(),
+        format!("Calling «bash -c 'which {}'» failed!", &program)
+    );
+
+    let location = String::from_utf8(output.stdout).expect(&format!(
+        "Found «{}», but the output is not utf8 clean.",
+        &program
+    ));
+
+    PathBuf::from(location.trim())
+}
+
+pub trait ProjectEnv {
+    fn write_to(&self, destination: &Path) -> Result<(), std::io::Error>;
+}
+
+pub struct ProjectEnvBuilderV1 {
+    env_vars: HashMap<OsString, OsString>,
+}
+
+impl ProjectEnvBuilderV1 {
+    pub fn new() -> ProjectEnvBuilderV1 {
+        ProjectEnvBuilderV1 {
+            env_vars: HashMap::new(),
+        }
+    }
+
+    pub fn set<S: Into<OsString>>(mut self, env_var: S, value: S) -> Self {
+        self.env_vars.insert(env_var.into(), value.into());
+        self
+    }
+}
+
+impl ProjectEnv for ProjectEnvBuilderV1 {
+    fn write_to(&self, destination: &Path) -> Result<(), std::io::Error> {
+        let output = Command::new(find_program("bash"))
```

Comment:
I don’t think you need `find_program` at all, from the [Command::new](https://doc.rust-lang.org/std/process/struct.Command.html?search=#method.new) docs:

> If program is not an absolute path, the PATH will be searched in an OS-defined way.

Reply:
Since we `env_clear`, we have to resolve it ourselves (or set the PATH). I added a comment to this effect in ec08cf2